### PR TITLE
fix: the pattern for git-url and make ref optional

### PR DESCRIPTION
**Pull request by swarnimarun**

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 Glob = NonEmptyStr
 UnsignedInt = conint(ge=0)
 GitUrl = constr(
-    pattern=r"/((git|ssh|http(s)?)|(git@[\w\.]+))(:(\/\/)?)([\w\.@\:\/\-~]+)(\.git)(\/)?/igm"
+    pattern=r"/((git|ssh|http(s)?)|(git@[\\w\\.]+))(:(//)?)([\\w\.@\:/\\-~]+)/igm"
```
Comment (swarnimarun) [inline model.py:20]:
```suggestion
    pattern=r"/((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\\w\.@\:/\\-~]+)/igm"
```

File: model.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ class GitBranch(BaseGitSource):
     branch: NonEmptyStr = Field(..., description="Branch to check out")
 
 
-GitSource = GitRev | GitTag | GitBranch
+GitSource = GitRev | GitTag | GitBranch | None
```
Comment (swarnimarun) [inline model.py:112]:
```suggestion
GitSource = GitRev | GitTag | GitBranch | BaseGitSource
```

File: model.py [hunk 3]
```diff
@@ -109,7 +109,7 @@ class GitBranch(BaseGitSource):
     branch: NonEmptyStr = Field(..., description="Branch to check out")
 
 
-GitSource = GitRev | GitTag | GitBranch
+GitSource = GitRev | GitTag | GitBranch | BaseGitSource
```
Comment (baszalmstra) [inline model.py:112]: All types derive from `BaseGitSource` so its not required here. Or can you also have a situation where you dont specify `rev`, `tag`, or `branch` at all?

Comment (swarnimarun) [reply model.py:112]: yep there can be, I assumed it meant latest head. At least there was in the tests that @wolfv wrote for the PR.

Comment (baszalmstra) [reply model.py:110]: 👍

**Pull request merged**